x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/DownsampleStep.java

```diff
@@ -35,9 +34,20 @@ public class DownsampleStep extends AsyncActionStep {
     private static final Logger logger = LogManager.getLogger(DownsampleStep.class);
 
     private final DateHistogramInterval fixedInterval;
+    private final StepKey nextStepOnSuccess;
+    private final StepKey nextStepOnFailure;
+    private boolean downsampleFailed;
```

Comment:
Maybe also use SetOnce<Boolean> here like in CreateSnapshotStep?

Reply:
I wanted to avoid having to set it at every successful outcome.  So, boolean value is initially set to `false` and on failure I set it to `true`.  This way, ILM will by default proceed to the next step.

Reply:
But I do think we need to make this field `volatile`? Since It is being accessed from different threads (line 123?).

Reply:
Done in [c097cad](https://github.com/elastic/elasticsearch/pull/94965/commits/c097cad86c983b93306468f51843a507b055e625)